Tasks/DotNetCoreInstallerV1/versionFetcher.ts

```diff
@@ -0,0 +1,273 @@
+        let adjacentChannels = [];
+        this.releasesIndex["releases-index"].forEach(channel => {
+            if (channel["channel-version"].startsWith(`${versionParts.majorVersion}`)) {
+                adjacentChannels.push(channel);
+            }
+        });
+
+        return adjacentChannels;
+    }
+
+    private detectMachineOS(): string[] {
+        let osSuffix = [];
+        let scriptRunner: trm.ToolRunner;
+
+        try {
+            console.log(tl.loc("DetectingPlatform"));
+            if (tl.osType().match(/^Win/)) {
+                let escapedScript = path.join(this.getCurrentDir(), 'externals', 'get-os-platform.ps1').replace(/'/g, "''");
+                let command = `& '${escapedScript}'`
+
+                let powershellPath = tl.which('powershell', true);
+                scriptRunner = tl.tool(powershellPath)
+                    .line('-NoLogo -Sta -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -Command')
+                    .arg(command);
+            }
+            else {
+                let scriptPath = path.join(this.getCurrentDir(), 'externals', 'get-os-distro.sh');
+                this.setFileAttribute(scriptPath, "777");
+
+                scriptRunner = tl.tool(tl.which(scriptPath, true));
+            }
+
+            let result: trm.IExecSyncResult = scriptRunner.execSync();
+
+            if (result.code != 0) {
+                throw tl.loc("getMachinePlatformFailed", result.error ? result.error.message : result.stderr);
+            }
+
+            let output: string = result.stdout;
+
+            let index;
+            if ((index = output.indexOf("Primary:")) >= 0) {
+                let primary = output.substr(index + "Primary:".length).split(os.EOL)[0];
+                osSuffix.push(primary);
+                console.log(tl.loc("PrimaryPlatform", primary));
+            }
+
+            if ((index = output.indexOf("Legacy:")) >= 0) {
+                let legacy = output.substr(index + "Legacy:".length).split(os.EOL)[0];
+                osSuffix.push(legacy);
+                console.log(tl.loc("LegacyPlatform", legacy));
+            }
+
+            if (osSuffix.length == 0) {
+                throw tl.loc("CouldNotDetectPlatform");
+            }
+        }
+        catch (ex) {
+            throw tl.loc("FailedInDetectingMachineArch", ex)
+        }
+
+        return osSuffix;
+    }
+
+    private setFileAttribute(file: string, mode: string): void {
+        fs.chmodSync(file, mode);
+    }
+
+    private getCurrentDir(): string {
+        return __dirname;
+    }
+
+    private releasesIndex: any;
+    private httpCallbackClient: httpClient.HttpClient;
+}
+
+export class VersionInfo {
+    public version: string;
+    public files: VersionFilesData[];
+
+    public static getRuntimeVersion(versionInfo: VersionInfo): string {
+        if (versionInfo["runtime-version"]) {
+            return versionInfo["runtime-version"];
+        }
+
+        return versionInfo.version;
```

Comment:
why are we returing this? Should we throw exception if the above field is not found

Reply:
This is to handle both cases 
1. `Package Type = sdk` : in this case the object contains `runtime-version` and that will be returned. This is needed to create completeFile for runtime installed along with sdk.
2. `PackageType = runtime`: in this case `version` itself is runtime-version, and thus that should be returned.

I feel that if runtime-version is not found in sdk object, we can skip creating the runtime complete file and give a warning. Changing code to behave the same.
@sachinma  thoughts ?